compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -215,8 +215,25 @@ public boolean checkForUniqueSymbol(Location pos, SymbolEnv env, BSymbol symbol)
                 dlog.error(pos, DiagnosticErrorCode.UNSUPPORTED_REMOTE_METHOD_NAME_IN_SCOPE, name);
                 return false;
             }
-
-            dlog.error(pos, DiagnosticErrorCode.REDECLARED_SYMBOL, name);
+            if (symbol.kind == SymbolKind.CONSTANT && symbol.type.tag == TypeTags.FINITE) {
+                BConstantSymbol symbolConst = (BConstantSymbol) symbol;
+                BConstantSymbol foundSymConst = (BConstantSymbol) foundSym;
+                BLangLiteral symbolTypeLiteral = (BLangLiteral) ((BFiniteType)
+                        symbolConst.type).getValueSpace().iterator().next();
+                String symbolValue = (symbolTypeLiteral.value == null) ?
+                        symbolTypeLiteral.originalValue : String.valueOf(symbolTypeLiteral.value);
+                BLangLiteral foundSymTypeLiteral = (BLangLiteral) ((BFiniteType)
+                        foundSymConst.type).getValueSpace().iterator().next();
+                String foundSymValue = (foundSymTypeLiteral.value == null) ?
+                        foundSymTypeLiteral.originalValue : String.valueOf(foundSymTypeLiteral.value);
+                int symbolLiteralType = symbolConst.literalType.tag;
+                int foundLiteralSymType = foundSymConst.literalType.tag;
+                if ((!symbolValue.equals(foundSymValue)) || (symbolLiteralType != foundLiteralSymType)) {
+                    dlog.error(pos, DiagnosticErrorCode.REDECLARED_SYMBOL, name);
+                }
+            } else {
+                dlog.error(pos, DiagnosticErrorCode.REDECLARED_SYMBOL, name);
+            }
```

Comment:
Should we prevent `BLangNodeTransformer` from adding duplicate top-level nodes and constants for duplicate enum members rather than handling the error msg here? @MaryamZi @gimantha WDYT?

Reply:
> ```ballerina
> const FIVE = 5;
> const FIVE = 5;
> ```
> 
> compiles without any error

@SandaruJayawardana  please check this

Reply:
> Should we prevent `BLangNodeTransformer` from adding duplicate top-level nodes and constants for duplicate enum members rather than handling the error msg here? @MaryamZi @gimantha WDYT?

How do we do this? by maintaining a global map or something?

Reply:
Yeh, something like that.

Reply:
Okay. I'll fix it.

Reply:
Since we need to evaluate the constant expressions for identifying the reinitialized enum members, I moved this check into 'ConstantValueResolver'.

Reply:
IMO, since enum members are a sugar of constants. Having a second initializer even with the same value should not be allowed. We can special case the default case though. @rdhananjaya @gimantha @MaryamZi WDYT?

Reply:
Let's clarify from James: https://github.com/ballerina-platform/ballerina-spec/issues/830#issuecomment-893999569